biorbd_optim/gui/plot.py

```diff
@@ -359,30 +406,36 @@ def __update_axes(self):
             p.set_ydata((np.nan, np.nan))
 
         for key in self.axes:
-            for i, ax in enumerate(self.axes[key][1]):
-                if not self.axes[key][0].ylim:
-                    y_max = -np.inf
-                    y_min = np.inf
-                    for p in ax.get_children():
-                        if isinstance(p, lines.Line2D):
-                            y_min = min(y_min, np.min(p.get_ydata()))
-                            y_max = max(y_max, np.max(p.get_ydata()))
-                    if np.isnan(y_min) or np.isinf(y_min):
-                        y_min = 0
-                    if np.isnan(y_max) or np.isinf(y_max):
-                        y_max = 1
-                    data_mean = np.mean((y_min, y_max))
-                    data_range = y_max - y_min
-                    if np.abs(data_range) < 0.8:
-                        data_range = 0.8
-                    y_range = (1.25 * data_range) / 2
-                    y_range = data_mean - y_range, data_mean + y_range
-                    ax.set_ylim(y_range)
-                    ax.set_yticks(np.arange(y_range[0], y_range[1], step=data_range / 4))
+            if not self.use_bounds_as_ylimit:
+                for i, ax in enumerate(self.axes[key][1]):
+                    if not self.axes[key][0].ylim:
+                        y_max = -np.inf
+                        y_min = np.inf
+                        for p in ax.get_children():
+                            if isinstance(p, lines.Line2D):
+                                y_min = min(y_min, np.min(p.get_ydata()))
+                                y_max = max(y_max, np.max(p.get_ydata()))
+                            y_range, data_range = self.__compute_ylim(y_min, y_max, 1.25)
+                            ax.set_ylim(y_range)
+                            ax.set_yticks(np.arange(y_range[0], y_range[1], step=data_range / 4,))
 
         for p in self.plots_vertical_lines:
             p.set_ydata((0, 1))
 
+    @staticmethod
+    def __compute_ylim(min_val, max_val, threshold):
+        if np.isnan(min_val) or np.isinf(min_val):
+            min_val = 0
+        if np.isnan(max_val) or np.isinf(max_val):
+            max_val = 1
+        data_mean = np.mean((min_val, max_val))
```

Comment:
I think there should be a test here to make sure min_val is not greater than max_val (in the cas min_val is nan and max_val is less than 0) and the same for max_val being smaller than min_val